commit 68b4512ca6fcd474a8e144863f03410ed186232b
Author: Nicholas Mc Guire <hofrat@osadl.org>
Date:   Wed Nov 28 12:24:47 2018 +0100

    perf: arm_spe: handle devm_kasprintf() failure
    
    [ Upstream commit 754a58db6a556e6e5f5e32f3e84e7d67b5bf9c8e ]
    
    devm_kasprintf() may return NULL on failure of internal allocation
    thus the assignment to 'name' is not safe if unchecked. If NULL
    is passed in for name then perf_pmu_register() would not fail
    but rather silently jump to skip_type which is not the intent
    here. As perf_pmu_register() may also return -ENOMEM returning
    -ENOMEM in the (unlikely) failure case of devm_kasprintf() should
    be fine here as well.
    
    Acked-by: Mark Rutland <mark.rutland@arm.com>
    Fixes: d5d9696b0380 ("drivers/perf: Add support for ARMv8.2 Statistical Profiling Extension")
    Signed-off-by: Nicholas Mc Guire <hofrat@osadl.org>
    [will: reworded error message]
    Signed-off-by: Will Deacon <will.deacon@arm.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/perf/arm_spe_pmu.c b/drivers/perf/arm_spe_pmu.c
index 54ec278d2fc4..e1a77b2de78a 100644
--- a/drivers/perf/arm_spe_pmu.c
+++ b/drivers/perf/arm_spe_pmu.c
@@ -927,6 +927,11 @@ static int arm_spe_pmu_perf_init(struct arm_spe_pmu *spe_pmu)
 
 	idx = atomic_inc_return(&pmu_idx);
 	name = devm_kasprintf(dev, GFP_KERNEL, "%s_%d", PMUNAME, idx);
+	if (!name) {
+		dev_err(dev, "failed to allocate name for pmu %d\n", idx);
+		return -ENOMEM;
+	}
+
 	return perf_pmu_register(&spe_pmu->pmu, name, -1);
 }
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1075_68b4512ca6fcd474a8e144863f03410ed186232b.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1075_68b4512ca6fcd474a8e144863f03410ed186232b.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
